src/mesh.F90

```diff
@@ -290,34 +290,42 @@ pure function mesh_intersects_2d(m, xyz0, xyz1) result(intersects)
 
     ! Check if line intersects left surface -- calculate the intersection point
     ! y
-    yi = y0 + (xm0 - x0) * (y1 - y0) / (x1 - x0)
-    if (yi >= ym0 .and. yi < ym1) then
-      intersects = .true.
-      return
+    if ( (x0 < xm0 .and. x1 > xm0) .or. (x0 > xm0 .and. x1 < xm0) ) then
```

Comment:
Are you sure it's too much whitespace?  I think it helps to visually separate the two distinct chunks of logic.

This logic is for correctness.  This accounts for the case of a line segment that points towards a mesh, but lies completely outside of it.  Essentially, the old `mesh_intersects_*` would extrapolate the line segment to a line of infinite length, and see if that infinite line intersects the mesh.  It's hard to describe... If you can't visualize what I'm describing, stop by my office and I can draw a picture.